Selection fixes

Dimitri0xff opened the pull request:
* added possibility to try selecting / deselecting in StageTest by clicking the labels
* some fixes
* you need only the 3 latest commits, the others are just to sync my fork with the origin, so I think you should cherry-pick the latest 3 commits (I messed up earlier by committing on my master branch)

StrongJoshua commented:
Can you squash the commits and resolve the conflicts? I'm sure i'ts just formatting (use the libGDX formatter, it's included in this repo).

Dimitri0xff commented:
Updated

* Commit 1: Somehow you duplicated a block of code during the merges
* Commit 2: My original changes rebased and squashed
* Commit 3: Your identation was like: 1 tab 1 space, or 2 tabs 2 spaces etc... I don't even know how you did that

StrongJoshua commented:
Merged. Should go live tonight if I'm not too busy.
